src/template_parser.h

```diff
@@ -444,6 +451,26 @@ class TemplateParser : public LexerHelper
 
             m_currentBlockInfo.range.startOffset = FinishCurrentBlock(matchStart);
             break;
+        case RM_RawBegin:
+            if (m_currentBlockInfo.type == TextBlockType::RawBlock)
+                break;
+            else if (m_currentBlockInfo.type != TextBlockType::RawText && m_currentBlockInfo.type != TextBlockType::Comment)
+            {
+                FinishCurrentLine(match.position() + 12);
```

Comment:
No sure we can use here the length as a constant. raw/endraw (according to regex) can contain spaces between the keyword and `%}`